internal/tracker/tracker.go

```diff
@@ -226,17 +226,43 @@ func (t *Tracker) trackImageReferences(ctx context.Context, urls []string, fresh
 	return nil
 }
 
-func (t *Tracker) trackGitReferences(_ context.Context, urls []string) error {
+func (t *Tracker) trackGitReferences(ctx context.Context, urls []string, freshen bool) error {
 	effective_on := effectiveOn()
 
+	if freshen {
+		log.Debug("Freshen is enabled")
+
+		for u := range t.TrustedTasks {
+			if strings.HasPrefix(u, "git+") {
+				urls = append(urls, u)
```

Comment:
If `urls` has a capacity greater than it's length, this call will actually change the caller's `urls` slice.

Reply:
I need to address this, it _might_ be fine, but I need to ponder this a bit

Reply:
I think it is okay, for two reasons: the `urls` will be a copy of the passed in `urls`, so the length of it will change here, but it will not be reflected in the in the scope of the `Track` function, i.e. in the `gitUrls` variable. The underlying array might get additional elements, if they do end up using the same array, but `url` in this function, and the `gitUrls` in the `Track` function will have their own lengths; the second reason is that the `gitUrls` in the `Track` function will not be used after it is passed to `trackGitReferences` so even if it did change there is no side effect of it.
Does it cause problems for the reader, perhaps not aware of this nuance? -- It _might_, so I'll refactor this just to make it easier for the reader...